worker/fanconfigurer/fanconfigurer.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package fanconfigurer
+
+import (
+	"fmt"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/juju/errors"
+	"github.com/juju/loggo"
+	"github.com/juju/utils/clock"
+
+	"github.com/juju/juju/network"
+	"github.com/juju/juju/utils/scriptrunner"
+	"github.com/juju/juju/watcher"
+	"github.com/juju/juju/worker/catacomb"
+)
+
+var logger = loggo.GetLogger("juju.worker.fanconfigurer")
+
+type FanConfigurer struct {
+	catacomb catacomb.Catacomb
+	config   FanConfigurerConfig
+	clock    clock.Clock
+	mu       sync.Mutex
+	enabled  bool
+}
+
+type FanConfigurerFacade interface {
+	FanConfig() (network.FanConfig, error)
+	WatchForFanConfigChanges() (watcher.NotifyWatcher, error)
+}
+
+type FanConfigurerConfig struct {
+	Facade FanConfigurerFacade
+}
+
+// processNewConfig acts on a new fan config.
+func (fc *FanConfigurer) processNewConfig() error {
+	logger.Debugf("Processing new fan config")
+	fc.mu.Lock()
+	defer fc.mu.Unlock()
+
+	fanConfig, err := fc.config.Facade.FanConfig()
+	if err != nil {
+		return err
+	}
+	if len(fanConfig) == 0 {
+		logger.Debugf("Fan not enabled")
+		// TODO(wpk) 2017-08-05 We have to clean this up!
+		return nil
```

Comment:
I was just thinking that if you ever had fan config, doesn't this cause it to remain when it is finally removed?

Reply:
It does, that's why there's a TODO - I don't yet know how to clean it 'cleanly'.